File: app/src/main/java/com/sky/chowder/ui/adapter/MainAdapter.java
package com.sky.chowder.ui.adapter;

import android.animation.AnimatorInflater;
import android.graphics.Outline;
import android.os.Build;
import android.support.annotation.RequiresApi;
import android.view.View;
import android.view.ViewOutlineProvider;

import com.sky.adapter.RecyclerAdapter;
import com.sky.adapter.RecyclerHolder;
import com.sky.chowder.R;
import com.sky.chowder.model.ActivityModel;

/**
 * @author sky QQ:1136096189
 * @Description:
 * @date 15/12/9 下午8:52
 */
public class MainAdapter extends RecyclerAdapter<ActivityModel> {

    public MainAdapter(int layoutId) {
        super(layoutId);
    }

    protected void onAchieveHolder(RecyclerHolder holder, int position) {
        holder.setText(R.id.tv_name, position + 1 + "." + datas.get(position).getClassName());
        holder.setText(R.id.tv_describe, datas.get(position).getDescribe());
        holder.setImage(R.id.img_describe, datas.get(position).getImg());
        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT_WATCH) {
            //使用ripple
            //((CardView) holder.getView(R.id.cardView)).setRadius(new Random().nextInt(50));
            //((CardView) holder.getView(R.id.cardView)).setCardElevation(new Random().nextInt(50));
            holder.getView(R.id.cardView).setBackground(context.getDrawable(R.drawable.ripple));
            //点击效果，阴影效果
            //((CardView) holder.getView(R.id.cardView)).setCardElevation(new Random().nextInt(100));
            holder.getView(R.id.cardView).setStateListAnimator(
                    AnimatorInflater.loadStateListAnimator(context, R.drawable.state_list_animator)
            );
            //视图裁剪
            holder.getView(R.id.img_describe).setClipToOutline(true);
            holder.getView(R.id.img_describe).setOutlineProvider(new ViewOutlineProvider() {
                @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
                @Override
                public void getOutline(View view, Outline outline) {
                    outline.setRoundRect(view.getLeft(), view.getTop(),
                            view.getRight(), view.getBottom(), 60);
                }
            });
        } else{
            holder.getView(R.id.cardView).setBackground(context.getResources().getDrawable(R.drawable.bg_card));
        }
    }

}

File: app/src/main/java/com/sky/chowder/ui/adapter/MainAdapter.kt
package com.sky.chowder.ui.adapter

import android.graphics.Typeface
import android.view.ViewGroup
import android.view.animation.LayoutAnimationController
import android.view.animation.ScaleAnimation
import com.sky.adapter.RecyclerAdapter
import com.sky.adapter.RecyclerHolder
import com.sky.chowder.R
import com.sky.chowder.model.ActivityModel
import kotlinx.android.synthetic.main.adapter_main.view.*

/**
 * Created by SKY on 2015/12/6.
 * 主页
 */
class MainAdapter(layoutId: Int) : RecyclerAdapter<ActivityModel>(layoutId) {
    var fontIcon = intArrayOf(R.string.font, R.string.font01, R.string.font02, R.string.font03,
            R.string.font04, R.string.font05, R.string.font06, R.string.font07, R.string.font08)

    override fun onAchieveHolder(holder: RecyclerHolder, position: Int) {
        //应写成异步调用
        val scale = ScaleAnimation(0f, 1f, 0f, 1f)
        scale.duration = 100
        if (position % 2 == 1) {
            val controller = LayoutAnimationController(scale, 0.5f)
            controller.order = LayoutAnimationController.ORDER_RANDOM
            (holder?.itemView as ViewGroup).layoutAnimation = controller
        } else holder?.itemView.startAnimation(scale)
        val face = Typeface.createFromAsset(context.assets, "font/icomoon.ttf")
        with(holder!!.itemView) {
            tvName.text = "${datas[position].className}" + resources.getString(fontIcon[position % 9])
            tvDescribe.text = resources.getString(fontIcon[8 - position % 9]) + datas[position].describe
            tvImage.text = resources.getString(fontIcon[position % 9])

            tvName.typeface = face
            tvDescribe.typeface = face
            tvImage.typeface = face
            tvImage.textSize=50f
        }
    }
}